eva-accession-import-dbsnp2/src/main/java/uk/ac/ebi/eva/accession/dbsnp2/io/ContigToGenbankReplacerProcessor.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2019 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.ac.ebi.eva.accession.dbsnp2.io;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.batch.item.ItemProcessor;
+import uk.ac.ebi.eva.accession.core.ClusteredVariant;
+import uk.ac.ebi.eva.accession.core.contig.ContigMapping;
+import uk.ac.ebi.eva.accession.core.contig.ContigSynonyms;
+import uk.ac.ebi.eva.accession.core.persistence.DbsnpClusteredVariantEntity;
+
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Converts the contig to its GenBank synonym when possible. If the synonym can't be determined it keeps the contig as
+ * is
+ */
+public class ContigToGenbankReplacerProcessor
```

Comment:
@jmmut @sundarvenkata-EBI should [ContigReplacerProcessor](https://github.com/EBIvariation/eva-accession/blob/master/eva-accession-import/src/main/java/uk/ac/ebi/eva/accession/dbsnp/processors/ContigReplacerProcessor.java) (from import module) be used as reference instead of [ContigToGenbankReplacerProcessor](https://github.com/EBIvariation/eva-accession/blob/master/eva-accession-pipeline/src/main/java/uk/ac/ebi/eva/accession/pipeline/steps/processors/ContigToGenbankReplacerProcessor.java) (pipeline module)?

Reply:
The one from the import module deals with SubsnpNoHGVS and its attributes as opposed to IVariant (which is what we have for this pipeline) which the pipeline module uses .

Reply:
I'm talking more about the business logic that deals with the assembly report, which is considerably more complex in `ContigReplacerProcessor`, but that is necessary to get dbSNP 1.0 data correctly imported.

Reply:
@108krohan Could you check if you can adapt the logic from https://github.com/EBIvariation/eva-accession/blob/master/eva-accession-import/src/main/java/uk/ac/ebi/eva/accession/dbsnp/processors/ContigReplacerProcessor.java for this processor? the attributes may differ since that processor uses SubSNPNoHGVS object but could you check if you can supply the analogous attributes from IVariant?

Reply:
As per my assessment, there doesn't seem to be any added benefit of using `ContigToGenbankReplacerProcessor` - as `DbsnpClusteredVariantEntity` object being processed does not have any `Chromosome` field - it has a `Contig` attribute only. Because this is so, the code that checks `Chromosome` attribute is not required. The business logic of `ContigToGenbankReplacerProcessor` dealing with `Contig` replacement with `Genbank` accession is near identical to the one being used in `ContigReplacerProcessor`.

So I would suggest that `ContigReplacerProcessor` be used as reference.

Reply:
I took a look and looks to me like [isContigReplaceable](https://github.com/EBIvariation/eva-accession/blob/master/eva-accession-import/src/main/java/uk/ac/ebi/eva/accession/dbsnp/processors/ContigSynonymValidationProcessor.java#L87) is pretty much equivalent to [isGenbankReplacementPossible](https://github.com/EBIvariation/eva-accession/blob/master/eva-accession-core/src/main/java/uk/ac/ebi/eva/accession/core/contig/ContigMapping.java#L163). 

My only opinion is that ContigNotFound should lead to Exception as opposed to just a warning.

Reply:
sorry for the delay answering this. In general, `ContigMapping.isGenbankReplacementPossible`  (which is used in the accession pipeline) should be used to replace to genbank. I confirm that ContigReplacerProcessor mainly focuses on handling both fields contig and chromosome, but also is more permissive on what replacements can be done in a corner case.

Looking at the code it seems you started copying the structure of ContigToGenbankReplacerProcessor and change it to process the approppiate model, using `ContigMapping.isGenbankReplacementPossible`, which is exactly what I recommend doing.

As sundar says, I would also throw an exception if a replacement is not possible. This allows to detect any unmet assumption we might have about human contigs.